script.js

```diff
@@ -1,4 +1,314 @@
+}
+
+function getScore(hand) {
+  let score = 0;
+  let aceCounter = 0;
+  for (let i = 0; i < hand.length; i++) {
+    if (hand[i].rank > 1 && hand[i].rank < 11) {
+      score += hand[i].rank;
+    } else if (hand[i].rank > 10) {
+      score += 10;
+    } else {
+      score += 11;
+      aceCounter += 1;
+    }
+  }
+
+  while (score > 21 && aceCounter > 0) {
+    score -= 10;
+    aceCounter -= 1;
+  }
+
+  return score;
+}
+
+function flipCard(hand, cardNo) {
+  var tempHand = hand;
+  tempHand[cardNo - 1].flipped = !tempHand[cardNo - 1].flipped;
+  return tempHand;
+}
+
+//Game Statement Generators
+
+function getWinner(playerHand, computerHand) {
+  let playerScore = getScore(playerHand);
+  let computerScore = getScore(computerHand);
+  if (
+    playerScore === 21 ||
+    (playerScore > computerScore && playerScore <= 21) ||
+    (computerScore > 21 && playerScore <= 21)
+  ) {
+    return "player";
+  } else if (
+    computerScore === 21 ||
+    (computerScore > playerScore && computerScore <= 21) ||
+    (playerScore > 21 && computerScore <= 21)
+  ) {
+    return "dealer";
+  } else if (computerScore === playerScore) {
+    return "draw";
+  } else {
+    return "";
+  }
+}
+
+function winStatement(playerHand, computerHand) {
+  let playerScore = getScore(playerHand);
+  let computerScore = getScore(computerHand);
+  let winner = getWinner(playerHand, computerHand);
+  let result = `<div id="result"`;
+
+  if (winner === "player") {
+    result += `style = "background-color: #B5EA8D">`;
+  } else if (winner === "dealer") {
+    result += `style = "background-color: #FF8484">`;
+  } else {
+    result += ">";
+  }
+
+  if (playerScore === 21 || computerScore === 21) {
+    result += "Blackjack! ";
+  } else if (playerScore > 21 || computerScore > 21) {
+    result += "Bust! ";
+  }
+
+  if (winner == "draw") {
+    result += "It's a draw!";
+  } else if (winner == "player") {
+    result += "You win!";
+  } else if (winner == "dealer") {
+    result += "Dealer wins!";
+  }
+  result += `</div>`;
+  return result;
+}
+
+function getSitrep() {
+  let result = `Your hand: ${displayHand(playerHand)}<br>Your score: ${getScore(
+    playerHand
+  )} <br><br>——————————<br><br> Dealer's hand: ${displayHand(
+    computerHand
+  )}<br>Dealer's score: `;
+
+  if (computerHand[1].flipped) {
+    result += "?";
+  } else {
+    result += getScore(computerHand);
+  }
+
+  return result;
+}
+
+// Card Aesthetics
+
+function displayHand(hand) {
+  let result = `<div id = "all-cards">`;
+  for (let i = 0; i < hand.length; i++) {
+    result += `<div id="${getCardColor(hand[i])}"> ${getCardName(
+      hand[i]
+    )} </div>`;
+  }
+  result += `</div>`;
+  return result;
+}
+
+function getCardColor(card) {
+  if (card.flipped === true) {
+    return "flipped-card";
+  }
+
+  if (card.suit === "hearts" || card.suit === "diamonds") {
+    return "redcard";
+  } else {
+    return "card";
+  }
+}
+
+function getCardName(card) {
```

Comment:
For majority of the logic here (other than the flipped card), you could actually just add a name key in the `makeDeck` function.